packages/components/src/Components/InsightsLabel/InsightsLabel.js

```diff
@@ -0,0 +1,55 @@
+import './labels.scss';
+
+import React, { useEffect, useState } from 'react';
+
+import { AngleDoubleDownIcon } from '@patternfly/react-icons';
+import { AngleDoubleUpIcon } from '@patternfly/react-icons';
+import CriticalIcon from './CriticalIcon';
+import { EqualsIcon } from '@patternfly/react-icons';
+import { Label } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+
+const VALUE_TO_STATE = {
+    1: { icon: <AngleDoubleDownIcon/>, text: 'Low' },
+    2: { icon: <EqualsIcon/>, text: 'Moderate' },
+    3: { icon: <AngleDoubleUpIcon/>, text: 'High' },
+    4: { icon: <CriticalIcon/>, text: 'Critical' }
+};
+
+const InsightsLabel = ({ value, text, hideIcon, labelProps }) => {
+
+    const [ icon, setIcon ] = useState();
```

Comment:
Why do you use state to store the icon text and icon? 

It seems as an overhead to me. The component will be rendered at least twice on initial render (render, setState, setState... the state could be stored as one value when it’s so tightly coupled..)

You can compute the props at render time:

```jsx
InsightsLabel = (props) => <Label
         { ...labelProps }
         className={`ins-c-label-${value}`}
         icon={!hideIcon && VALUE_TO_STATE[value].icon}
     >
         { text || VALUE_TO_STATE[value].text }
     </Label>;
```